src/textual/file_monitor.py

```diff
@@ -12,7 +11,7 @@
 class FileMonitor:
     """Monitors a file for changes and invokes a callback when it does."""
 
-    def __init__(self, path: str | PurePath, callback: Callable) -> None:
+    def __init__(self, path: PurePath, callback: Callable) -> None:
```

Comment:
Is this class going to remain part of the public API? If so, I can change this back to give users flexibility.

If it's just going to be internal though I think we should convert all str | PurePath at the API boundary so we have less propagation of unions like this.

Reply:
This probably should be private. I agree about reducing unions.